### PR TITLE
Add XML comments in ZooKeeperBasedMembershipTable

### PR DESCRIPTION
**Pull request by benjaminpetit**
I just copied comments from the interface, since the documentation doesn't support inheritance :(

### CODE REVIEW

File: src/OrleansZooKeeperUtils/ZooKeeperBasedMembershipTable.cs [hunk 1]
```diff
@@ -55,6 +55,11 @@ public class ZooKeeperBasedMembershipTable : IMembershipTable, IGatewayListProvi
 
         private TimeSpan maxStaleness;
 
+        /// <summary>
+        /// Initializes the ZooKeeper based gateway provider
+        /// </summary>
+        /// <param name="config">The configuration for this instance.</param>
```
Comment (xiazen) [inline src/OrleansZooKeeperUtils/ZooKeeperBasedMembershipTable.cs]: The configuration for this client ? 

what do you mean by instance?

File: src/OrleansZooKeeperUtils/ZooKeeperBasedMembershipTable.cs [hunk 2]
```diff
@@ -172,6 +224,18 @@ public Task<bool> UpdateRow(MembershipEntry entry, string etag, TableVersion tab
...
+        /// Updates the IAmAlive part (column) of the MembershipEntry for this silo.
+        /// This operation should only update the IAmAlive collumn and not change other columns.
+        /// This operation is a "dirty write" or "in place update" and is performed without etag validation. 
+        /// With regards to eTags update:
+        /// This operation may automatically update the eTag associated with the given silo row, but it does not have to. It can also leave the etag not changed ("dirty write").
+        /// With regards to TableVersion:
+        /// this operation should not change the TableVersion of the table. It should leave it untouched.
+        /// There is no scenario where this operation could fail due to table semantical reasons. It can only fail due to network problems or table unavailability.
+        /// </summary>
+        /// <param name="entry"></param>
```
Comment (xiazen) [inline src/OrleansZooKeeperUtils/ZooKeeperBasedMembershipTable.cs]: fill in the param?

Comment (xiazen) [inline src/OrleansZooKeeperUtils/ZooKeeperBasedMembershipTable.cs]: wow never tried the reivew feature before. kind of cool

**Comment by benjaminpetit** [comment]
@dotnet-bot test this please

**Comment by xiazen** [comment]
LGTM. will push this after the dotnet-bot tested it.

**Pull request merged**